ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+
+        if (MMIsEmptyString(hMMMD.aLayerName))
+            return 0;  // If no file, no error. Just continue.
+
+        memcpy(&hMMMD.hBB, &hMiraMonLayer->TopHeader.hBB, sizeof(hMMMD.hBB));
+        hMMMD.pLayerDB = hMiraMonLayer->pLayerDB;
+        hMMMD.aArcFile = strdup_function(
+            get_filename_function(hMiraMonLayer->MMPolygon.MMArc.pszLayerName));
+        nResult = MMWriteMetadataFile(&hMMMD);
+        free_function(hMMMD.aArcFile);
+        return nResult;
+    }
+    else if (layerPlainType == MM_LayerType_Node)
+    {
+        // Node from arc
+        if (layerMainPlainType == MM_LayerType_Arc)
+        {
+            hMMMD.aLayerName = hMiraMonLayer->MMArc.MMNode.pszREL_LayerName;
+            if (MMIsEmptyString(hMMMD.aLayerName))
+                return 0;  // If no file, no error. Just continue.
+            memcpy(&hMMMD.hBB, &hMiraMonLayer->MMArc.TopNodeHeader.hBB,
+                   sizeof(hMMMD.hBB));
+        }
+        else  // Node from polygon
+        {
+            hMMMD.aLayerName =
+                hMiraMonLayer->MMPolygon.MMArc.MMNode.pszREL_LayerName;
+            if (MMIsEmptyString(hMMMD.aLayerName))
+                return 0;  // If no file, no error. Just continue.
+            memcpy(&hMMMD.hBB,
+                   &hMiraMonLayer->MMPolygon.MMArc.TopNodeHeader.hBB,
+                   sizeof(hMMMD.hBB));
+        }
+        hMMMD.pLayerDB = nullptr;
+        return MMWriteMetadataFile(&hMMMD);
+    }
+    return 0;
+}
+
+int MMWriteVectorMetadata(struct MiraMonVectLayerInfo *hMiraMonLayer)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (hMiraMonLayer->bIsPoint)
+        return MMWriteVectorMetadataFile(hMiraMonLayer, MM_LayerType_Point,
+                                         MM_LayerType_Point);
+    if (hMiraMonLayer->bIsArc && !hMiraMonLayer->bIsPolygon)
+    {
+        if (MMWriteVectorMetadataFile(hMiraMonLayer, MM_LayerType_Node,
+                                      MM_LayerType_Arc))
+            return 1;
+        return MMWriteVectorMetadataFile(hMiraMonLayer, MM_LayerType_Arc,
+                                         MM_LayerType_Arc);
+    }
+    if (hMiraMonLayer->bIsPolygon)
+    {
+        if (MMWriteVectorMetadataFile(hMiraMonLayer, MM_LayerType_Node,
+                                      MM_LayerType_Pol))
+            return 1;
+        if (MMWriteVectorMetadataFile(hMiraMonLayer, MM_LayerType_Arc,
+                                      MM_LayerType_Pol))
+            return 1;
+        return MMWriteVectorMetadataFile(hMiraMonLayer, MM_LayerType_Pol,
+                                         MM_LayerType_Pol);
+    }
+    if (hMiraMonLayer->bIsDBF)
+    {
+        return MMWriteVectorMetadataFile(hMiraMonLayer, MM_LayerType_Unknown,
+                                         MM_LayerType_Unknown);
+    }
+    return 0;
+}
+
+// Verifies the version of a MiraMon REL 4 file.
+int MMCheck_REL_FILE(char *szREL_file)
```

Comment:
I suspect it could be const?
```suggestion
int MMCheck_REL_FILE(const char *szREL_file)
```

Reply:
Sure, Done.